### ISSUE TITLE
`a11y-link-in-text-block` rule should avoid flagging Links with `className`

### ISSUE THREAD

**Issue by khiga8** [primer/eslint-plugin-primer-react#192]
## Context

The `a11y-link-in-text-block` rule isn't accounting for how a `Link` may already have sufficient styling applied through a `class`.

For example:

```.html
<Link href="#" className='some-special-class'>@khiga8</Link> posted in discussion.
```

It is not possible for the lint rule to know what styles are being applied via `some-class`. If `some-class` applies bold styling, then technically speaking, it doesn't need an `inline` prop.

We may want to skip `Link` that have `className` entirely to reduce potential false positives.
